webcrypto/subtle_crypto.go

```diff
@@ -137,8 +141,87 @@ func (sc *SubtleCrypto) Verify(
 //   - SHA-512
 //
 // The `data` parameter should contain the data to be digested.
-func (sc *SubtleCrypto) Digest(algorithm goja.Value, data interface{}) *goja.Promise {
-	return nil
+func (sc *SubtleCrypto) Digest(algorithm interface{}, data goja.Value) *goja.Promise {
+	promise, resolve, reject := sc.makeHandledPromise()
+
+	// Validate that the value we received is either an ArrayBuffer, TypedArray, or DataView
+	// This uses the technique described in https://github.com/dop251/goja/issues/379#issuecomment-1164441879
+	var (
+		isArrayBuffer = IsInstanceOf(sc.vu.Runtime(), data, ArrayBufferConstructor)
+		isDataView    = IsInstanceOf(sc.vu.Runtime(), data, DataViewConstructor)
+		isTypedArray  = IsTypedArray(sc.vu.Runtime(), data)
+	)
+
+	if !isArrayBuffer && !isDataView && !isTypedArray {
+		reject(fmt.Errorf("data must be an ArrayBuffer, TypedArray, or DataView"))
```

Comment:
Here (and in other places below), we don't need `fmt.Errorf`. It's just a `errors.New`

Reply:
Good catch 👍🏻